orc8r/cloud/go/obsidian/swagger/spec_servicer.go

```diff
@@ -30,32 +26,34 @@ type specServicer struct {
 	standaloneSpec string
 }
 
-// NewSpecServicerFromFile intializes a spec servicer
-// given a service name.
-func NewSpecServicerFromFile(service string) protos.SwaggerSpecServer {
-	service = strings.ToLower(service)
-	partialPath, standalonePath := getSpecPaths(service)
-	partial, err := ioutil.ReadFile(partialPath)
+// NewSpecServicer constructs a spec servicer.
+func NewSpecServicer(partialSpec string, standaloneSpec string) protos.SwaggerSpecServer {
+	return &specServicer{partialSpec: partialSpec, standaloneSpec: standaloneSpec}
+}
+
+func NewSpecServicerFromSpecs(specs Specs, service string) protos.SwaggerSpecServer {
+	// Swallow errors because the service should continue to run even if it
+	// can't find its Swagger spec Specs.
+	partial, err := specs.GetPartialSpec(service)
 	if err != nil {
-		// Swallow error because the service should continue to
-		// run even if it can't find its partial Swagger spec file.
-		glog.Errorf("Error retrieving Swagger Spec of service %s: %+v", service, err)
-		return NewSpecServicer("", "")
+		glog.Errorf(
+			"Error retrieving Swagger partial spec of service %s: %+v",
+			service,
+			err)
 	}
-	standalone, err := ioutil.ReadFile(standalonePath)
+	standalone, err := specs.GetStandaloneSpec(service)
 	if err != nil {
-		// Swallowing ReadFile error because the service should continue to
-		// run even if it can't find its standalone Swagger spec file.
-		glog.Errorf("Error retrieving Swagger Spec of service %s: %+v", service, err)
-		return NewSpecServicer("", "")
+		glog.Errorf(
```

Comment:
Here and elsewhere: don't line-break these. We're fine with 120+ char lines, and even past that just wrap as non-aggressively as possible

Reply:
Curiously, there appears to be a convention around 80 col wrap on comments. One might consider that the format of source code is also for human consumption, which might lead us to consider the same 80 col wrap for both?

Reply:
So (a) the doc comments get ingested by external tools and could end up anywhere and (b) they will naturally end up a lot longer than 80/120 chars if we don't hard-break

To me the goal is to get a natural / easy-to-read flow. And line-breaking code decreases readability imo, while line-breaking docs doesn't have the same effect because a "line" is not a semantically meaningful designation in the latter

But ultimately, the Go convention (e.g. go internal code) is ~120 chars for code, 80 chars for docs